packages/core/src/parser/magic-anchor-parser.ts

```
@@ -0,0 +1,180 @@
+import { type Result, success, failure } from '../lib/result.js';
+import { makeError } from '../lib/error.js';
+import { parseInputSchema } from '../schemas/index.js';
+import { fromZod } from '../lib/zod-adapter.js';
+
+/**
+ * Parser for Magic Anchor syntax (`:A: marker prose`).
+ * Handles both sync and async parsing with Result pattern support.
+ */
+export class MagicAnchorParser {
+  // :A: api regex pattern for matching anchor syntax
+  private static readonly ANCHOR_PATTERN = /:A: ([^]*?)(?=\n|$)/g;
```

Biome (1.9.4)

[error] The regular expression includes this negated empty character class.

Negated empty character classes match anything.
If you want to match against [, escape it \[.
Otherwise, remove the character class or fill it.

(lint/correctness/noEmptyCharacterClassInRegex)